docs/spec/sources.md

```diff
@@ -362,5 +375,41 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects). It
+is highly-recommended producing the `knsource` CloudEvent extension.
+
+The `knsource` value SHOULD be of the form `<plural>.<group>/<name>`, where:
+
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `name` is the [Source Custom Object](#source-custom-objects) name
+
+Examples:
+
+- `pingsources.eventing.knative.dev/hello`
+- `rabbitmq.sources.knative.dev/rabbitmq-source`
+
+It is allowed to specify an alternative value in
+[`spec.ceOverrides.extensions`](#duckspec), in which case the `knsource` value
+can be of any shape.
+
+When produced, the `knsource` CloudEvent extension MUST appear in
+[status.ceAttributes](#duckstatus).
```

Comment:
How? `CloudEventAttributes` contains only `Type` and `Source`.

Reply:
oh good catch! That's right, not possible. I removed this statement for now and I'll open an issue.